sql/src/main/java/io/crate/execution/engine/join/HashInnerJoinBatchIterator.java

```diff
@@ -156,7 +155,14 @@ public boolean moveNext() {
         return true;
     }
 
-    private void updateBlockSize() {
+    private void resetBuffer() {
+        calculateBlockSize();
+        this.buffer = new IntObjectHashMap<>(this.blockSize);
+        numberOfRowsInBuffer = 0;
+
+    }
+
+    private void calculateBlockSize() {
```

Comment:
minor: I'd prefer that this method returns the blockSize instead of also setting it as this would also fit better to the methods name